packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -494,12 +495,20 @@ async function getResourceManagementClientForArmDeployment(
 
 async function compileBicepToJson(
   bicepCommand: string,
-  bicepOrchestrationFilePath: string
+  bicepOrchestrationFilePath: string,
+  logger: LogProvider | undefined
 ): Promise<JSON> {
-  const command = `${bicepCommand} build "${bicepOrchestrationFilePath}" --stdout`;
   try {
-    const result = await Executor.execCommandAsync(command);
-    return JSON.parse(result.stdout as string);
+    const result = await executeCommand(
+      undefined,
+      logger,
+      { shell: false },
+      bicepCommand,
```

Comment:
When `shell: false`, you may also need to change bicep checker to remove the `"` in 
```
return `"${this.getBicepExecPath()}"`;
```
in `bicepChecker.ts`